internal/healthcheck/healthcheck.go

```diff
@@ -102,12 +117,18 @@ func (c *Check) HandleReadiness(w http.ResponseWriter, _ *http.Request) {
 	w.Write([]byte("ok"))
 }
 
-func ready(err error, minReady uint64) bool {
+func ready(err error, minReady, total int) bool {
 	mErr, ok := err.(proxy.MultiErr)
 	if !ok {
 		return false
 	}
-	return minReady != 0 && minReady <= uint64(len(mErr))
+	if minReady == 0 {
+		return false
+	}
```

Comment:
Why a special case for minReady? Shouldn't areReady always >= minReady if min is 0?

Reply:
I'll remove this.

Reply:
Checking for 0 is necessary, because it means the min-ready query param was not set and so any error is a failure for the check. I'll add a comment.

Reply:
Added support for 0, which always returns a 200. Also, if a caller specifies more than the number of registered instances, returns a 503.

Reply:
And after writing that, I've changed my mind. Min-ready must be greater than zero. Otherwise it's a 400.